bootstrap/config/config_test.go

```diff
@@ -92,3 +108,237 @@ func TestGetSecretPathsChanged(t *testing.T) {
+			nil, false, []byte("false"), nil, nil, errors.New("configuration provider is not available")},
+		{"Invalid - common config not ready", &serviceConfig, config.ServiceTypeOther, nil,
+			nil, true, []byte("false"), nil, nil, errors.New("common config is not loaded")},
+		{"Invalid - common config not ready error", &serviceConfig, config.ServiceTypeOther, nil,
+			nil, true, []byte("false"), errors.New("config provider not available"), nil, errors.New("failed to retrieve common config value IsCommonConfigReady: config provider not available")},
+		{"Valid - core service", &serviceConfig, config.ServiceTypeOther, nil,
+			nil, true, []byte("true"), nil, errors.New("configuration not found"), errors.New("failed to load the common configuration for all services: configuration not found")},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.Name, func(t *testing.T) {
+			// create parameters for the processor
+			f := flags.New()
+			f.Parse(nil)
+			mockLogger := logger.MockLogger{}
+			env := environment.NewVariables(mockLogger)
+			timer := startup.NewStartUpTimer(serviceKey)
+			ctx, cancel := context.WithCancel(context.Background())
+
+			wg := sync.WaitGroup{}
+			dic := di.NewContainer(di.ServiceConstructorMap{
+				container.LoggingClientInterfaceName: func(get di.Get) interface{} { return mockLogger },
+			})
+			// create the processor
+			proc := NewProcessor(f, env, timer, ctx, &wg, nil, dic)
+			// set up mocks
+			providerClientMock := &mocks.Client{}
+			providerClientCreator := func(logger.LoggingClient,
+				string,
+				string,
+				types.GetAccessTokenCallback,
+				types.ServiceConfig) (configuration.Client, error) {
+				return providerClientMock, tc.providerClientErr
+			}
+			providerClientMock.On("IsAlive").Return(tc.isAlive)
+			serviceConfigMock := ConfigurationMockStruct{}
+			if tc.isAlive {
+				providerClientMock.On("GetConfigurationValue", "IsCommonConfigReady").Return(tc.isCommonConfigReady, tc.commonConfigReadyErr)
+			}
+			ccReady, err := strconv.ParseBool(string(tc.isCommonConfigReady))
+			require.NoError(t, err)
+			if ccReady {
+				providerClientMock.On("GetConfiguration", &serviceConfigMock).Return(tc.serviceConfig, tc.getConfigErr).Once()
+			}
+			if tc.serviceType == config.ServiceTypeApp || tc.serviceType == config.ServiceTypeDevice {
+				providerClientMock.On("GetConfiguration", &serviceConfigMock).Return(tc.serviceTypeConfig, tc.getConfigErr).Once()
+			}
+			// call load common config
+			err = proc.loadCommonConfig(common.ConfigStemAll, getAccessToken, &ProviderInfo{}, &serviceConfigMock, tc.serviceType, providerClientCreator)
+			// make assertions
+			providerClientMock.AssertExpectations(t)
+			require.NotNil(t, cancel)
+			if tc.expectedErr == nil {
+				assert.NoError(t, err)
+				assert.NotNil(t, serviceConfigMock.Writable.LogLevel)
+				switch tc.serviceType {
+				case config.ServiceTypeApp:
+					assert.False(t, serviceConfigMock.Writable.StoreAndForward.Enabled)
```

Comment:
Since false is the default `zero`  value, the mock should use `true` and check for `true` here.